aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -53,29 +53,77 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
     }
 
     // This function is called by the BlockExecutor for each transaction is intends
-    // to execute (via the ExecutorTask trait). It can be as a part of sequential
-    // execution, or speculatively as a part of a parallel execution.
-    fn execute_transaction(
+    // to execute (via the ExecutorTask trait). This function is run speculatively
+    // as a part of a parallel execution.
+    fn execute_transaction_parallel(
+        &self,
+        view: &impl StateView,
+        txn: &PreprocessedTransaction,
+        txn_idx: TxnIndex,
+        aggregator_enabled: bool,
+    ) -> ExecutionStatus<AptosTransactionOutput, VMStatus> {
+        let log_context = AdapterLogSchema::new(self.base_view.id(), txn_idx as usize);
+        match self.vm.execute_single_transaction(
+            txn,
+            &view.as_move_resolver(),
+            &log_context,
+            aggregator_enabled,
+        ) {
+            Ok((vm_status, vm_output, sender)) => {
+                // If aggregators are disabled for user transactions, then delta_change_set should be empty as there should not be any delta outputs.
+                assert!(
+                    aggregator_enabled
+                        || !matches!(txn, PreprocessedTransaction::UserTransaction(_))
+                        || vm_output.delta_change_set().is_empty()
+                );
+                if vm_output.status().is_discarded() {
```

Comment:
This is shared with `execute_transaction_sequential`. Let's refactor this out?

Reply:
Done.

Reply:
I did push. Are you looking at the right commit version? Please take a look at the latest one (https://github.com/aptos-labs/aptos-core/pull/8334/files#diff-a8984b647508e3ad3ec3c2aee50ac6aff300d263fabf5b25d357fcbae2dc70c9).